File: src/Application Specific/Classes/Entity Collections/ContractOption3/ContractOption3Assets/ContractOption3SiteAssetValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace ContractOption3SiteAsset
    {
        // make sure that contact object is valid
        public class ContractOption3SiteAssetValidator : BaseValidator
        {
            public void Validate(ContractOption3SiteAsset oContractOption3SiteAsset)
            {

                // make sure that contact object is valid
                if (oContractOption3SiteAsset.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oContractOption3SiteAsset.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/ContractOption3/ContractOption3Assets/ContractOption3SiteAssetValidator.vb
Namespace Entity

    Namespace ContractOption3SiteAsset

        Public Class ContractOption3SiteAssetValidator
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oContractOption3SiteAsset As ContractOption3SiteAsset)

                'make sure that contact object is valid
                If oContractOption3SiteAsset.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oContractOption3SiteAsset.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If



                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

        End Class

    End Namespace

End Namespace
